debezium-core/src/main/java/io/debezium/transforms/outbox/EventRouter.java

```diff
@@ -81,6 +81,9 @@ public R apply(R r) {
         // Ignoring messages which do not adhere to the CDC Envelope, for instance:
         // Heartbeat and Schema Change messages
         if (!smtManager.isValidEnvelope(r)) {
+            if (smtManager.isHeartbeat(r)) {
```

Comment:
@Naros I wonder if we should use nested conditions. Would not it be better to introduce new method on SMT manager like `isDataChangeMessage`? It might make sense in case of other heartbeat sensitive SMTs.

Reply:
+1, I like that idea.